pkg/awsutils/awsutils.go

```diff
@@ -685,14 +691,30 @@ func awsUtilsErrInc(fn string, err error) {
 	awsUtilsErr.With(prometheus.Labels{"fn": fn, "error": err.Error()}).Inc()
 }
 
+func exponentialBackoff(retryFunc func() error) error {
+	settings := backoff.NewExponentialBackOff()
+	settings.MaxInterval = exponentialBackoffMaxInterval
+	settings.MaxElapsedTime = exponentialBackoffMaxElapsedTime
+
+	return backoff.Retry(retryFunc, backoff.WithMaxRetries(settings, exponentialBackoffMaxRetries))
+}
+
 // FreeENI detachs ENI interface and delete ENI interface
 func (cache *EC2InstanceMetadataCache) FreeENI(eniName string) error {
 	log.Infof("Trying to free eni: %s", eniName)
 
 	// find out attachment
 	//TODO: use metadata
-	_, attachID, err := cache.DescribeENI(eniName)
-	if err != nil {
+	var attachID *string
+	var err error
+	describeEniFunc := func() error {
+		_, attachID, err = cache.DescribeENI(eniName)
+		if err != nil {
```

Comment:
should probably add some better error type handling here (and in all the retried functions), as there are some errors that we should not retry (and potentially treat as successes, depending on the operation)

For example, in the `DetachNetworkInterface` call (we should experiment with this), if AWS successfully detaches, but fails to report that back to us, a later call may return the error `InvalidAttachmentID.NotFound`
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#CommonErrors